enhancements/baremetal/enable-baremetal-on-other-platforms.md

```diff
@@ -0,0 +1,170 @@
+---
+title: enable-baremetal-on-other-platforms
+authors:
+  - "@asalkeld"
+  - "@sadasu"
+reviewers:
+  - TBD
+  - "@hardy"
+  - "@romfreiman"
+  - "@dhellman"
+approvers:
+  - TBD
+creation-date: 2021-08-20
+last-updated: 2021-08-20
+status: implementable
+see-also:
+  - "/enhancements/baremetal-provisiong-config.md"
+replaces:
+superseded-by:
+---
+
+# Enable Baremetal on other Platforms to support ZTP
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Baremetal Host API is only available when deploying an OpenShift cluster with the baremetal
+platform (via the IPI or AI (Assisted Installer) workflow). Having the ability to
+manage baremetal hosts from clusters without requiring the cluster to be on baremetal
+would be beneficial to customers.
+
+## Motivation
+
+An initial driver of this feature are the centralized host management use cases
+in edge topologies, which without this feature, is restricted to having the
+central OpenShift cluster deployed on baremetal.
+
+### Goals
+
+The specific goals of this proposal are to:
+
+Support the ZTP (Zero Touch Provisioning) use case by partially enabling Baremetal Host API
+on the following platforms:
+- None
+- OpenStack
+- vSphere
+
+We will be successful when:
+
+ZTP can deploy clusters when running on the above platforms.
+
+### Non-Goals
+
+Allow Baremetal Host API to be fully enabled on all platforms.
+
+Note that the scope of this epic is only enablement of the Metal3
+BaremetalHost API and associated controller/services - the edge ZTP flow
+currently interacts directly via this API without any Machine API integration.
+
+## Proposal
+
+BMO (baremetal-operator) provides the Baremetal Host API, it in turn is configured
+and managed by CBO (cluster-baremetal-operator).
+
+CBO reads the Provisioning CR that is created by the installer on baremetal platforms
```

Comment:
Can you please elaborate how will it look in SNO case (none platform). Who will create the provisioning CR? What will change in the current behavior?

Reply:
On baremetal + AI:
assisted-service (in internal/ignition/ignition.go) generates the ignition - part of that is the Provisioning CR
If it is not on baremetal then that will need to be created by what ever system is driving the flow (as the installer will not create it). I'd assume Assisted Installer or Hive?

Reply:
hmm not sure. if it's a bare metal, then there is no problem already.
for ztp - not sure AI is the place. Who's currently configures the bmo to watch all namesapces? I assume it's not AI but part of the installation guide.  
oc patch provisioning provisioning-configuration --type merge -p '{"spec":{"watchAllNamespaces": true}}'
So i guess it's also up to the user / ztp scritps?

Reply:
Yes, I think we can say it's part of the installation instructions but that it doesn't have to be automated.

Reply:
thanks, I'll add a note to this effect.